miner/scroll_worker.go

```diff
@@ -465,18 +465,24 @@ func (w *worker) startNewPipeline(timestamp int64) {
 	if w.chainConfig.CurieBlock != nil && w.chainConfig.CurieBlock.Cmp(header.Number) == 0 {
 		misc.ApplyCurieHardFork(parentState)
 
+		var nextL1MsgIndex uint64
+		if dbVal := rawdb.ReadFirstQueueIndexNotInL2Block(w.eth.ChainDb(), header.ParentHash); dbVal != nil {
+			nextL1MsgIndex = *dbVal
+		}
+
```

Comment:
instead of duplicating this here, we can move the query at L517 to before this Curie fork check.

Reply:
True, but I'd rather keep it as is for this version and refactor later (since we've already used this way for Sepolia).